### DISCUSSION TITLE
SABnzbd fails to install python "This environment is externally managed"

### DISCUSSION THREAD

**Discussion by bobloadmire** [tteck/Proxmox#3288] [Help Needed]
`SABnzbd-4.3.2/interfaces/Glitter/templates/include_messages.tmpl
SABnzbd-4.3.2/interfaces/Glitter/templates/include_queue.tmpl
SABnzbd-4.3.2/interfaces/Glitter/templates/include_overlays.tmpl
SABnzbd-4.3.2/interfaces/Glitter/templates/main.tmpl
SABnzbd-4.3.2/interfaces/Glitter/templates/static/bootstrap/css/bootstrap.min.css
SABnzbd-4.3.2/interfaces/Glitter/templates/static/bootstrap/js/bootstrap.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/bootstrap/fonts/glyphicons-halflings-regular.eot
SABnzbd-4.3.2/interfaces/Glitter/templates/static/bootstrap/fonts/glyphicons-halflings-regular.ttf
SABnzbd-4.3.2/interfaces/Glitter/templates/static/bootstrap/fonts/glyphicons-halflings-regular.svg
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/glitter.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/jquery-ui.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/glitter.filelist.pagination.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/glitter.main.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/date.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/search-query-parser.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/jquery.hotkeys.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/knockout-3.5.1.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/jquery-3.5.1.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/glitter.basic.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/glitter.history.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/moment-2.26.0.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/knockout-extensions.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/jquery.peity.min.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/glitter.queue.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/pl.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/fi.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/ru.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/he.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/nb.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/zh-cn.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/pt-br.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/da.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/de.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/sv.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/fr.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/nl.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/sr.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/es.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/javascripts/momentjs_locale/ro.js
SABnzbd-4.3.2/interfaces/Glitter/templates/static/stylesheets/glitter.css
SABnzbd-4.3.2/interfaces/Glitter/templates/static/stylesheets/glitter.mobile.css
SABnzbd-4.3.2/interfaces/Glitter/templates/static/stylesheets/colorschemes/Auto.css
SABnzbd-4.3.2/interfaces/Glitter/templates/static/stylesheets/colorschemes/Night.css
SABnzbd-4.3.2/interfaces/Glitter/templates/static/stylesheets/colorschemes/Light.css
SABnzbd-4.3.2/interfaces/Config/README.txt
SABnzbd-4.3.2/interfaces/Config/templates/config_notify.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/_inc_header_uc.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_switches.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_rss.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_cat.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_special.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_folders.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_server.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_scheduling.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/_inc_footer_uc.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_sorting.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config_general.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/main.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/config.tmpl
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/bootstrap/css/bootstrap.min.css
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/bootstrap/js/bootstrap.min.js
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/bootstrap/fonts/glyphicons-halflings-regular.eot
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/bootstrap/fonts/glyphicons-halflings-regular.ttf
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/bootstrap/fonts/glyphicons-halflings-regular.svg
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/css/chartist.min.css
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/css/Auto.css
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/css/login.css
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/css/Night.css
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/css/style.css
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/images/logo-arrow.svg
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/images/logo-full.svg
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/images/logo-small.svg
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/js/jquery-ui.min.js
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/js/jquery-3.5.1.min.js
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/js/script.js
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/js/filesize.min.js
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/js/jquery.tablesort.min.js
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/js/chartist.min.js
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/safari-pinned-tab.svg
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/favicon.ico
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/apple-touch-icon-152x152-precomposed.png
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/faviconpaused.ico
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/android-192x192.png
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/apple-touch-icon-180x180-precomposed.png
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/apple-touch-icon-120x120-precomposed.png
SABnzbd-4.3.2/interfaces/Config/templates/staticcfg/ico/apple-touch-icon-76x76-precomposed.png
SABnzbd-4.3.2/interfaces/Config/templates/login/main.tmpl
SABnzbd-4.3.2/interfaces/wizard/inc_bottom.tmpl
SABnzbd-4.3.2/interfaces/wizard/index.html
SABnzbd-4.3.2/interfaces/wizard/inc_top.tmpl
SABnzbd-4.3.2/interfaces/wizard/two.html
SABnzbd-4.3.2/interfaces/wizard/one.html
SABnzbd-4.3.2/interfaces/wizard/README.TXT
SABnzbd-4.3.2/interfaces/wizard/static/style.css
SABnzbd-4.3.2/interfaces/wizard/static/javascript/checkserver.js
 /error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.

[ERROR] in line 43: exit code 0: while executing command $STD python3 -m pip install -r requirements.txt`

Have no idea why i'm getting this error. advanced install with debian 12 and root enabled, verbose enabled. failed multiple attempts. automatic default install also failed.

**Comment by tteck** [comment]
Fixed

**Reply by bobloadmire** [reply]
confirmed, thanks.

**Reply by tteck** [reply]
Debian 12.6 brought more challenges. 😉
